examples/complete/main.tf

```diff
@@ -1,105 +1,111 @@
 
-public_subnet_ids = ["subnet-0f6fc716d9cd6f6d9", "subnet-09a8bf8febe10cbe5"]
+  private_subnet_ids = data.aws_subnets.private.ids
 
-control_plane_subnet_ids = ["subnet-07e1f1c21c6f47689", "subnet-0cdfd688b6a439e77","subnet-0f6fc716d9cd6f6d9", "subnet-09a8bf8febe10cbe5"]
+  public_subnet_ids = data.aws_subnets.public.ids
 
-create_eks = true
+  control_plane_subnet_ids = concat(data.aws_subnets.private.ids, data.aws_subnets.public.ids)
 
-cluster_timeouts = {
-  "create": "20m"
-  "update": "20m"
-  "delete": "30m"
-}
+  create_eks = var.create_eks
 
-cluster_name = "arc-blueprint"
+  cluster_timeouts = var.cluster_timeouts
+  cluster_name     = var.cluster_name
 
-cluster_version = "1.24"
+  cluster_version = var.cluster_version
 
-create_cluster_security_group = true
+  create_cluster_security_group = var.create_cluster_security_group
 
-cluster_security_group_name = "eks-cluster-sg"
+  cluster_security_group_name = var.cluster_security_group_name
 
-cluster_security_group_use_name_prefix = true
+  cluster_security_group_use_name_prefix = var.cluster_security_group_use_name_prefix
 
-cluster_security_group_description = "EKS cluster security group"
+  cluster_security_group_description = var.cluster_security_group_description
 
-cluster_security_group_id = null
+  cluster_security_group_id = var.cluster_security_group_id
 
-cluster_additional_security_group_ids = []
+  cluster_additional_security_group_ids = var.cluster_additional_security_group_ids
 
-cluster_security_group_additional_rules = {}
+  cluster_security_group_additional_rules = var.cluster_security_group_additional_rules
 
-cluster_security_group_tags = {
-  "Project": "eks-blueprint"
-}
+  cluster_security_group_tags = module.tags.tags
 
-create_cluster_primary_security_group_tags = true
+  create_cluster_primary_security_group_tags = var.create_cluster_primary_security_group_tags
 
-cluster_endpoint_public_access = true
+  cluster_endpoint_public_access = var.cluster_endpoint_public_access
 
-cluster_endpoint_private_access = false
+  cluster_endpoint_private_access = var.cluster_endpoint_private_access
 
-cluster_endpoint_public_access_cidrs = ["0.0.0.0/0"]
+  cluster_endpoint_public_access_cidrs = var.cluster_endpoint_public_access_cidrs
 
-cluster_kms_key_arn = null
+  cluster_kms_key_arn = var.cluster_kms_key_arn
 
-cluster_kms_key_deletion_window_in_days = 30
+  cluster_kms_key_deletion_window_in_days = var.cluster_kms_key_deletion_window_in_days
 
-cluster_kms_key_additional_admin_arns = []
+  cluster_kms_key_additional_admin_arns = var.cluster_kms_key_additional_admin_arns
 
-enable_cluster_encryption = true
+  enable_cluster_encryption = var.enable_cluster_encryption
 
-cluster_encryption_config = [{
-  "provider_key_arn": "arn:aws:kms:us-east-1:804295906245:key/6c70aa11-30e6-497f-ba68-fdeedb018dab",
-  "resources": ["secrets"]
-}]
+  cluster_encryption_config = [{
+    "provider_key_arn" : "arn:aws:kms:us-east-1:804295906245:key/6c70aa11-30e6-497f-ba68-fdeedb018dab",
```

Comment:
@rayl15  @jamescrowley321  

This is CMK so should there be a resource for key should be added instead of hardcoding so that CMK will get created first and then it will go to eks. 
Suggestions needed

Reply:
Can we just a built in key or does it have to be a CMK? If it has to be a CMK, use the CP module.

Reply:
@mayank0202 Make a GH issue for it and complete it in another PR.

Reply:
@jamescrowley321  I didn't find any built in key for that so thats why it seems like it needs to be CMK , yea will use CP module for that.